openquakeplatform/openquakeplatform/taxtweb/views.py

```diff
@@ -29,43 +29,107 @@
 from django.utils.text import slugify
 from django.contrib.messages.api import get_messages
 
-from openquakeplatform.vulnerability.models import TA, TYPES_OF_ASSESSMENT 
+def populate(s):
+    """
+function populate(s) {
+    var sar, subar, dirx, diry, el, i;
+
+    sar = s.split('/');
+
+    //
+    //  DIRECTION
+    dirx = acheck(sar, 'DX');
+
+    if (dirx == null || dirx == "DX" || dirx == "DX+D99") {
+        $("#Direction1RB1").prop("checked", true);
+        taxt_Direction1RB1Click(null);
+    }
+    else if (dirx == "DX+PF") {
+        $("#Direction1RB2").prop("checked", true);
+        taxt_Direction1RB2Click(null);
+    }
+    else {
+        alert("Not valid 'Direction X' found.");
+        return false;
+    }
+
+    diry = acheck(sar, 'DY');
+
+    // find sub-section for DirX material
+    if (dirx == null && diry == null) {
+        subar = sar;
+    }
+    else if (dirx == null && diry != null) {
+        alert("Invalid format: DY specified without DX.");
+        return false;
+    }
+    else {
+        dirx_idx =
+
+for (i = 0 ; i < sar.length ; i++) {
+        }
+
+    }
+    //
+    //  MATERIAL (Dir X)
+
+
+
+}
+    """
+
+    return True
 
 def index(request, **kwargs):
-    
     try:
-        type_of_assessment = int(request.GET.get("type_of_assessment",
-                                                 TA.FRAGILITY))
-        if type_of_assessment not in dict(TYPES_OF_ASSESSMENT):
-            type_of_assessment = TA.FRAGILITY
+        tab_id = int(request.GET.get("tab_id", 1))
+        if tab_id < 1 or tab_id > 4:
+            tab_id = 1
     except ValueError as e:
-        type_of_assessment = TA.FRAGILITY
+        tab_id = 1
 
-    desc = [ 'Structural System', 'Building Information', 'Exterior Attributes', 'Roof/Floor/foundation' ]
-    menu_content = ""
-    for i in range(0, len(desc)):
-        menu_content = (menu_content +
-                        '<li id="menu_id-%d" class="vuln_menu%s%s" onclick="menu_set(this);"><span>%s</span></li>' %
-                        (i+1, ("_selected" if i + 1 == type_of_assessment else ""), 
-                        (" vuln_menu_first" if i == 0 else ""), desc[i]))
+    try:
+        subtab_id = int(request.GET.get("subtab_id", 1))
+        if subtab_id < 1 or subtab_id > 2:
+            subtab_id = 1
+    except ValueError as e:
+        subtab_id = 1
 
+    if 'HTTP_HOST' in request.META:
+        proto = (request.META['HTTP_X_FORWARDED_PROTO'] if
+                 'HTTP_X_FORWARDED_PROTO' in request.META else 'http')
+        if request.META['HTTP_HOST'].startswith('taxtweb'):
+            taxt_prefix = proto + '://' + request.META['HTTP_HOST']
+        else:
+            taxt_prefix = proto + '://' + request.META['HTTP_HOST'] + '/taxtweb'
+    else:
+        taxt_prefix = "http://taxtweb.openquake.org"
```

Comment:
For the future I would suggest to move this value in the settings. I would prefer to have another configuration parameter rather then an hardcoded url in the code.